bitriseclient/bitriseclient.go

```diff
@@ -0,0 +1,939 @@
+
+	GetUploadedProvisioningProfileUUIDby(profileSlug string) (UUID string, err error)
+
+	RegisterProvisioningProfile(provisioningProfSize int64, profile profileutil.ProvisioningProfileInfoModel) (RegisterProvisioningProfileResponseData, error)
+
+	UploadProvisioningProfile(uploadURL string, uploadFileName string, outputDirPath string, exportFileName string) error
+
+	ConfirmProvisioningProfileUpload(profileSlug string, provUploadName string) error
+
+	FetchUploadedIdentities() ([]FetchUploadedIdentityListResponseData, error)
+
+	getUploadedIdentityDownloadURLBy(certificateSlug string) (downloadURL string, password string, err error)
+
+	downloadUploadedIdentity(downloadURL string) (content string, err error)
+
+	GetUploadedCertificatesSerialby(identitySlug string) (certificateSerialList []big.Int, err error)
+
+	RegisterIdentity(certificateSize int64) (RegisterIdentityResponseData, error)
+
+	UploadIdentity(uploadURL string, uploadFileName string, outputDirPath string, exportFileName string) error
+
+	ConfirmIdentityUpload(certificateSlug string, certificateUploadName string) error
+}
+
+// BitriseClient ...
+type BitriseClient struct {
+	accessToken     string
+	selectedAppSlug string
+}
+
+// New returns all the application of the user on Bitrise
+func (client *BitriseClient) New(accessToken string) (apps []Application, err error) {
+	client.accessToken = accessToken
+
+	log.Infof("Asking your application list from Bitrise...")
+
+	requestURL, err := urlutil.Join(baseURL, appsEndPoint)
+	if err != nil {
+		return []Application{}, err
+	}
+
+	log.Debugf("\nRequest URL: %s", requestURL)
+
+	headers := map[string]string{
+		"Authorization": "token " + client.accessToken,
+	}
+
+	request, err := createRequest("GET", requestURL, headers, map[string]interface{}{})
+	if err != nil {
+		return []Application{}, err
+	}
+
+	// Response struct
+	var appListResponse FetchMyAppsResponse
+	responseStatusCode := -1
+
+	//
+	// Perform request
+	if err := retry.Times(1).Wait(5 * time.Second).Try(func(attempt uint) error {
+		body, statusCode, err := performRequest(request)
+		if err != nil {
+			log.Warnf("Attempt (%d) failed, error: %s", attempt+1, err)
+			if !strings.Contains(err.Error(), "failed to perform request") {
+				log.Warnf("Response status: %d", statusCode)
+				log.Warnf("Body: %s", string(body))
+			}
+			return err
+		}
+
+		responseStatusCode = statusCode
+
+		// Parse JSON body
+		if err := json.Unmarshal([]byte(body), &appListResponse); err != nil {
+			return fmt.Errorf("failed to unmarshal response (%s), error: %s", body, err)
+		}
+		return nil
+
+	}); err != nil {
+		log.Errorf("Fetching list failed %s", err)
+		return []Application{}, err
+	}
+
+	// Success
+	log.Donef("Request succeeded with status code: %d", responseStatusCode)
+	logDebugPretty(appListResponse)
+
+	return appListResponse.Data, nil
+}
+
+// SetSelectedAppSlug ...
+func (client *BitriseClient) SetSelectedAppSlug(selectedAppSlug string) {
```

Comment:
This is a really small context for this variable, no need for this long name
selectedAppSlug string -> slug string